subprojects/workers/src/main/java/org/gradle/process/internal/daemon/WorkerDaemonManager.java

```diff
@@ -31,10 +31,10 @@
     private final MemoryManager memoryManager;
     private final WorkerDaemonExpiration workerDaemonExpiration;
 
-    public WorkerDaemonManager(WorkerDaemonClientsManager clientsManager, MemoryManager memoryManager, MemoryInfo memoryInfo) {
+    public WorkerDaemonManager(WorkerDaemonClientsManager clientsManager, MemoryManager memoryManager) {
         this.clientsManager = clientsManager;
         this.memoryManager = memoryManager;
-        this.workerDaemonExpiration = new WorkerDaemonExpiration(clientsManager, memoryInfo);
+        this.workerDaemonExpiration = new WorkerDaemonExpiration(clientsManager, TotalPhysicalMemoryProvider.getTotalPhysicalMemory());
```

Comment:
Beware that `getTotalPhysicalMemory()` might throw `UnsupportedOperationException`.